### ISSUE TITLE
orm: failed to detect mismatched column type

### ISSUE THREAD

**Issue by Dialga** [vlang/v#12425]
**V version:** Latest git
**OS:** Arch Linux x64

**What did you do?**
```
import sqlite
import json

struct Tiddler {
  id int
  created int
}

struct Tiddlers {
  id int [primary; sql: serial]
  created string
}

fn main() {
	tiddler := json.decode(Tiddler,'{}') or { Tiddler{} }
	db := sqlite.connect(':memory:') or { panic(err) }
	sql db {
		create table Tiddlers
		insert tiddler into Tiddlers
	}
}
```

**What did you expect to see?**
Compiled successfully
 
**What did you see instead?**
```
==================
/tmp/v_1000/ormfail.14399098660649273568.tmp.c:35213: error: cannot convert 'int' to 'struct string'
...
==================
(Use `v -cg` to print the entire error message)

builder error:
==================
C error. This should never happen.
```

In addition if `or { Tiddler{} }` is removed the v error is underlined for the wrong line
```
ormfail.v:16:2: error: assignment requires a non empty `or {}` block
   14 | fn main() {
   15 |     tiddler := json.decode(Tiddler,'{}')
   16 |     db := sqlite.connect(':memory:') or { panic(err) }
      |     ~~
   17 |     sql db {
   18 |       create table Tiddlers
```

**Comment by enghitalo** [comment]
```v
module main
import sqlite

import x.json2 as json
struct Tiddler {
  id int
  created int
}

struct Tiddlers {
  id int [primary; sql: serial]
  created string
}

fn main() {
	tiddler := json.decode[Tiddler]('{}') or { Tiddler{} }
	db := sqlite.connect(':memory:') or { panic(err) }
	sql db {
		create table Tiddlers
		insert tiddler into Tiddlers
	}
}
```
```bash
==================
/tmp/v_1000/v_bug.1893914730363162162.tmp.c:19525: error: cannot convert 'int' to 'struct string'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

**Comment by Dialga** [comment]
There was a fix for this in #12662, but never merged.

**Comment by enghitalo** [comment]
> There was a fix for this in #12662, but never merged.

Thanks
